dev/lib/syntax.js

```diff
@@ -77,17 +77,21 @@ function tokenizeTasklistCheck(effects, ok, nok) {
    * @type {State}
    */
   function inside(code) {
-    // Currently we match how GH works in files.
-    // To match how GH works in comments, use `markdownSpace` (`[\t ]`) instead
-    // of `markdownLineEndingOrSpace` (`[\t\n\r ]`).
+    // Accept space or tab as unchecked
     if (markdownLineEndingOrSpace(code)) {
       effects.enter('taskListCheckValueUnchecked')
       effects.consume(code)
       effects.exit('taskListCheckValueUnchecked')
       return close
     }
 
-    if (code === codes.uppercaseX || code === codes.lowercaseX) {
+    // Accept any other character (except ], newline, EOF) as checked
+    // This allows Obsidian-flavored checkboxes like [x], [X], [r], [!], etc.
+    if (
+      code !== codes.rightSquareBracket &&
+      code !== codes.eof &&
+      !markdownLineEnding(code)
```

Comment:
Both `- [[]` and `- []]` are invalid in Obsidian to avoid confusion with wikilink syntax.